ebpfsvc/svcmain.cpp

```diff
@@ -12,23 +12,23 @@ SERVICE_STATUS_HANDLE ebpf_service_status_handle;
 HANDLE ebpf_service_stop_event_handle = nullptr;
 
 void WINAPI
-service_control_handler(DWORD ctrl);
+service_control_handler(unsigned long ctrl);
 void
-service_report_event(PTSTR function);
+service_report_event(_Null_terminated_ wchar_t* function);
 void
-report_service_status(DWORD current_state, DWORD win32exitcode, DWORD wait_hint);
+report_service_status(unsigned long current_state, unsigned long win32exitcode, unsigned long wait_hint);
 void
-service_init(DWORD argc, PTSTR* argv);
+service_init(unsigned long argc, _Null_terminated_ wchar_t** argv);
```

Comment:
shouldn't ```argv``` be ```_In_z_ const``` here as it is 'input only' (with ```_Null_terminated_``` being used on _output_ pointers that point to null-terminated strings)?  We'd need to review other uses of the ```_Null_terminated_``` annotation in this PR for a similar scenario as well.

Reply:
I think it's not `_In_z_` since it's an array of strings, each of which is `_In_z_`. The array itself is not (and I think `_Null_terminated_` is incorrect for the same reason). Let me know if my understanding is incorrect.

Reply:
So  I guess just ```_In_ const``` would be more appropriate here (and elsewhere where we use such arrays of strings).  The fact still remains that this is still an _input_ _only_ parameter, right?

Reply:
`_In_reads_(argc)`

Reply:
> `_In_reads_(argc)`

even better! :-)